Add ParseOptions argument to standardSchemaV1

rehos opened the pull request:
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

Add optional ParseOptions argument to standardSchemaV1. This will make form validation in form libraries like [@tanstack/form](https://tanstack.com/form/latest) more usable (i.e. display all errors to the user instead of just the first)

gcanti commented on packages/effect/test/Schema/Schema/standardSchemaV1.test.ts:
```diff
@@ -2,6 +2,7 @@ import type { StandardSchemaV1 } from "@standard-schema/spec"
 import { Context, Effect, ParseResult, Predicate, Schema } from "effect"
 import { assertTrue, deepStrictEqual, strictEqual } from "effect/test/util"
 import { describe, it } from "vitest"
+import { b } from "vitest/dist/chunks/suite.BJU7kdY9.js"
```
this seems like an incorrect automatic import

rehos replied:
Sorry I will fix that

gcanti reviewed (changes requested):
I wonder if we should make `{ errors: "all" }` the default. AFAIK, the spec doesn't specify how many issues should be raised by default.

rehos commented:
I would prefer the default with `{ errors: "all" }`. Should we do something like:

```ts
export const standardSchemaV1 = <A, I>(
  schema: Schema<A, I, never>,
  overrideOptions?: AST.ParseOptions // Changed
): StandardSchemaV1<I, A> => {
  const decodeUnknown = ParseResult.decodeUnknown(schema, { errors: "all" }) // Changed
  return {
    "~standard": {
      version: 1,
      vendor: "effect",
      validate(value) {
        const scheduler = new scheduler_.SyncScheduler()
        const fiber = Effect.runFork(
          Effect.matchEffect(decodeUnknown(value, overrideOptions), { // Changed
            onFailure: makeStandardFailureFromParseIssue,
            onSuccess: (value) => Effect.succeed({ value })
          }),
          { scheduler }
        )
        scheduler.flush()
        const exit = fiber.unsafePoll()
        if (exit) {
          return makeStandardResult(exit)
        }
        return new Promise((resolve) => {
          fiber.addObserver((exit) => {
            resolve(makeStandardResult(exit))
          })
        })
      }
    }
  }
}

gcanti commented:
> Should we do something like:

Yes 👍

rehos commented:
@gcanti I have changed the default to all errors with the possibility override the parse options. Also added an extra test for the override use case with`{ errors: "first" }`. And the incorrect import has been fixed.

gcanti commented on .changeset/five-melons-leave.md:
```diff
@@ -0,0 +1,5 @@
+"effect": patch
+---
+
+Add ParseOptions argument to standardSchemaV1
```
```suggestion
Schema: `standardSchemaV1` now returns all errors by default and supports custom options.

The `standardSchemaV1` now returns **all validation errors** by default (`ParseOptions = { errors: "all" }`). Additionally, it now accepts an optional `overrideOptions` parameter, allowing you to customize the default parsing behavior as needed.
```

gcanti commented:
Thanks @rehos

rehos commented:
Thank you @gcanti for your quick response and feedback.
